library/psa_crypto.c

```diff
@@ -6142,15 +6084,42 @@ psa_status_t psa_generate_key( const psa_key_attributes_t *attributes,
     if( status != PSA_SUCCESS )
         goto exit;
 
+    /* In the case of a transparent key or an opaque key stored in local
+     * storage (thus not in the case of generating a key in a secure element
+     * or cryptoprocessor with storage), we have to allocate a buffer to
+     * hold the generated key material. */
+    if( slot->key.data == NULL )
+    {
+        if ( PSA_KEY_LIFETIME_GET_LOCATION( attributes->core.lifetime ) ==
```

Comment:
better to declare key_buffer_size here?

Reply:
Although `key_buffer_size` is only used in scope of this if statement I don't see it as a problem that it is declared above with the other declarations. In fact I think it could even make it harder to read and find the declarations and thus I prefer it the way it is.

Reply:
I will think about this a bit more and may address it in a follow-up PR.